parsl/tests/test_staging/test_zip_out.py

```diff
@@ -0,0 +1,99 @@
+import parsl
+import pytest
+import zipfile
+
+from parsl.data_provider.files import File
+from parsl.data_provider.data_manager import default_staging
+from parsl.data_provider.zip import ZipFileStaging
+
+from parsl.providers import LocalProvider
+from parsl.channels import LocalChannel
+from parsl.launchers import SimpleLauncher
+
+from parsl.config import Config
+from parsl.executors import HighThroughputExecutor
+
+from parsl.tests.configs.htex_local import fresh_config as local_config
+
+
+@pytest.mark.local
+def test_zip_path_split():
+    from parsl.data_provider.zip import zip_path_split
+    assert zip_path_split("/tmp/foo/this.zip/inside/here.txt") == ("/tmp/foo/this.zip", "inside/here.txt")
+
+
+@parsl.bash_app
+def output_something(outputs=()):
+    """This should output something into every specified output file:
+    the position in the output sequence will be written into the
+    corresponding output file.
+    """
+    cmds = []
+    for n in range(len(outputs)):
+        cmds.append(f"echo {n} > {outputs[n]}")
+
+    return "; ".join(cmds)
+
+
+@pytest.mark.local
+def test_zip_out(tmpd_cwd):
+    # basic test of zip file stage-out
+    zip_path = str(tmpd_cwd) + "container.zip"
+    file_base = "data.txt"
+    of = File("zip:" + zip_path + "/" + file_base)
```

Comment:
From the nitpicky dept, `tmpd_cwd` does the `/` for you.  Consider:
```python
zip_path = tmpd_cwd / "container.zip"
file_base = "data.txt"
of = File(f"zip:{zip_path / file_base}")
```

Note, though, that there is a slight difference between these two:
```python
str(tmpd_cwd) + "container.zip"  # --> /some/tmp/dircontainer.zip
str(tmpd_cwd / "container.zip")  # --> /some/tmp/dir/container.zip
```

So, mostly a bike-shed, and some other instances of same in this file, but a suggestion.